parmed/openmm/parameters.py

```diff
@@ -141,12 +145,18 @@ def write(self, dest, provenance=None):
             own_handle = True
         else:
             own_handle = False
+        typeified = False
         if self.atom_types:
             try:
                 self.typeify_templates()
+                typeified = True
             except KeyError:
                 warnings.warn('Some residue templates are using unavailable '
                               'AtomTypes')
+        if clean_params:
+            assert typeified, """Typification of the templates was not successful.
```

Comment:
Let's turn this into a warning.  Also, never use `assert` unless triggering the assert means a bug *in the library*.  The only times I use assert is when the design of ParmEd dictates that something must *always* be `True`.  For example, [here is an assertion that the combining rule must be either lorentz or geometric](https://github.com/ParmEd/ParmEd/blob/master/parmed/amber/_amberparm.py#L806-L807), but [setting it to anything else is an explicit error](https://github.com/ParmEd/ParmEd/blob/master/parmed/structure.py#L1381-L1385).

This is a defensive assertion to make sure that if I someday add support for another combining rule but forget to update the `AmberParm` subclass, an assertion error will hopefully limit the severity of the bug.  This is also true wherever I raise an `AssertionError` (which is ignored in the coverage calculation since I never WANT to hit those lines).  But most critically is the fact that [what Python means by "optimized compilation" is to get rid of assertions](http://stackoverflow.com/questions/2830358/what-are-the-implications-of-running-python-with-the-optimize-flag).  So the moral is, if you want an error check to be guaranteed to run, don't use `assert`.  It's there to aid in development and debugging.